pkg/search/proxy/store/resource_cache.go

```diff
@@ -1,5 +1,162 @@
+}
+
+func (c *resourceCache) stop() {
+	klog.Infof("Stop store for %s %s", c.cluster.Name, c.resource)
+	c.Store.DestroyFunc()
+}
+
+func newResourceCache(cluster *clusterv1alpha1.Cluster, gvr schema.GroupVersionResource, gvk schema.GroupVersionKind,
+	namespaced bool, newClientFunc func() (dynamic.NamespaceableResourceInterface, error)) (*resourceCache, error) {
+	s := &genericregistry.Store{
+		DefaultQualifiedResource: gvr.GroupResource(),
+		NewFunc: func() runtime.Object {
+			o := &unstructured.Unstructured{}
+			o.SetAPIVersion(gvk.GroupVersion().String())
+			o.SetKind(gvk.Kind)
+			return o
+		},
+		NewListFunc: func() runtime.Object {
+			o := &unstructured.UnstructuredList{}
+			o.SetAPIVersion(gvk.GroupVersion().String())
+			o.SetKind(gvk.Kind + "List")
+			return o
+		},
+		TableConvertor: rest.NewDefaultTableConvertor(gvr.GroupResource()),
+		// CreateStrategy tells whether the resource is namespaced.
+		CreateStrategy: restCreateStrategy(namespaced),
+		DeleteStrategy: restDeleteStrategy,
+	}
+
+	err := s.CompleteWithOptions(&generic.StoreOptions{
+		RESTOptions: &generic.RESTOptions{
+			StorageConfig: &storagebackend.ConfigForResource{
+				Config: storagebackend.Config{
+					Paging: utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking),
+					Codec:  unstructured.UnstructuredJSONScheme,
+				},
+				GroupResource: gvr.GroupResource(),
+			},
+			ResourcePrefix: gvr.Group + "/" + gvr.Resource,
+			Decorator:      storageWithCacher(newClientFunc, defaultVersioner),
+		},
+		AttrFunc: getAttrsFunc(namespaced),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &resourceCache{
+		cluster:  cluster,
+		Store:    s,
+		resource: gvr,
+	}, nil
+}
+
+func storageWithCacher(newClientFunc func() (dynamic.NamespaceableResourceInterface, error), versioner storage.Versioner) generic.StorageDecorator {
+	return func(
+		storageConfig *storagebackend.ConfigForResource,
+		resourcePrefix string,
+		keyFunc func(obj runtime.Object) (string, error),
+		newFunc func() runtime.Object,
+		newListFunc func() runtime.Object,
+		getAttrsFunc storage.AttrFunc,
+		triggerFuncs storage.IndexerFuncs,
+		indexers *cache.Indexers) (storage.Interface, factory.DestroyFunc, error) {
+		cacherConfig := cacherstorage.Config{
+			Storage:        newStore(newClientFunc, versioner, resourcePrefix),
+			Versioner:      versioner,
+			ResourcePrefix: resourcePrefix,
+			KeyFunc:        keyFunc,
+			GetAttrsFunc:   getAttrsFunc,
+			Indexers:       indexers,
+			NewFunc:        newFunc,
+			NewListFunc:    newListFunc,
+			Codec:          storageConfig.Codec,
+		}
+		cacher, err := cacherstorage.NewCacherFromConfig(cacherConfig)
+		if err != nil {
+			return nil, nil, err
+		}
+		return cacher, cacher.Stop, nil
+	}
+}
+
+var (
+	restCreateStrategyForNamespaced = &simpleRESTCreateStrategy{namespaced: true}
+	restCreateStrategyForCluster    = &simpleRESTCreateStrategy{namespaced: false}
+	restDeleteStrategy              = &simpleRESTDeleteStrategy{}
```

Comment:
Is this kind of processing really OK, doesn't cause panic when the method is called?
I ran the following test, but I can't guarantee that it will actually simulate the current situation:
```go
type AA interface {
	B()
}

type simpleAA struct {
	AA
}

var aa = &simpleAA{}

func main() {
	 aa.B()
}
```

Test result with:
```
panic: runtime error: invalid memory address or nil pointer dereference
```

Reply:
Their methods are not called in reading rest. 

Here assign `DeleteStrategy` in `genericregistry.Store` just to pass this statement:
https://github.com/karmada-io/karmada/blob/614e28508336d6c03a938ce1bf0678dafef034f0/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1320-L1322

And assign  `CreateStrategy` for statement:
https://github.com/karmada-io/karmada/blob/614e28508336d6c03a938ce1bf0678dafef034f0/vendor/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1310-L1318

Bu you are right, we shall explicitly declare others methods are not supporte.

Reply:
Okay